### ISSUE TITLE
Some UIA PropertyChanged events are not being generated

### ISSUE THREAD

**Issue by DaveTryon** [dotnet/winforms#7261]
### .NET version

.NET 6.0.5

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No

### Issue description

In https://github.com/microsoft/accessibility-insights-windows/issues/1346, we received a user report that apps listening for UIA PropertyChanged events on WinForm labels were being notified of changes in the element's BoundingRectangle, but not of changes of the element's Name. At first we thought this might be a problem with Accessibility Insights, but upon further investigation we confirmed that we receive the expected events from WPF applications. There is a sample app in the AIWin bug, but I've also created a gist at https://gist.github.com/DaveTryon/44421194268fe2f5fc6624c13553f117. The test app just uses a timer to update a label on 2 second intervals. 

### Steps to reproduce

1. Clone the gist at https://gist.github.com/DaveTryon/44421194268fe2f5fc6624c13553f117
1. Build the application
1. Run application
1. Using [Application Insights for Windows](https://github.com/microsoft/accessibility-insights-windows/releases/latest), hover over the element that shows the update count
1. In the hierarchy tree, click on the "..." icon next to the element and choose "Listen to Events"
1. Expand the Properties control and check Name (alternatively, just check next to Properties to enable all properties)
1. Click the red dot to begin listening to events
1. Watch for the events--no name events are generated

The events are also not detected by AccEvent.exe. Events from a similar WPF app are detected by both Accessibility Insights and AccEvent. We suspect that this is an issue in the WinForms UIA provider. It may apply more broadly just just label controls, but we made no attempt to identify a full set of impacted control types.

**Comment by Tanya-Solyanik** [comment]
As part of this work item, we need to investigate what controls other than `Label` don't raise property change event for `Name` property. @Olina-Zhang - would you be able to investigate that, Dave had provided a sample app that can be modified to use other controls.

**Comment by DaveTryon** [comment]
@Olina-Zhang, the GIF you provided shows only EventRecorderNotification events--the Name property in the Event details shows what property we're registering to receive. Once the registration completes, there should be an AutomationPropertyChanged event _each time_ the control text changes. Here's a GIF showing what happens using a WPF-based app:

![NameChangedEvents](https://user-images.githubusercontent.com/45672944/172729073-4319c558-825a-4e71-b8e6-88d02de03a7a.gif)

I ran the WinForms app provided by @Tanya-Solyanik, and none of the controls produced AutomationPropertyChanged events. That's expected on the NumberSpinner, since it doesn't automatically set the Name property when the numeric value changes.

**Comment by NikitaSemenovAkvelon** [comment]
In progress

**Comment by John-Qiao** [comment]
Verified with 7.0.100-rc.1.22378.6 from main branch, this issue has been fixed that it shows the AutomationPropertyChanged or Text_TextChanged event each time the control text changes, please check below screenshots:

![1](https://user-images.githubusercontent.com/45864985/181674924-dd9d0c9c-a46d-494c-b47c-e3b8600a07cb.png)
![2](https://user-images.githubusercontent.com/45864985/181674931-73f09a41-0e2e-4feb-9f4b-5a21ec7c25ec.png)
![3](https://user-images.githubusercontent.com/45864985/181674934-a2bd914c-dd2e-4c6a-908e-959a5c63cb89.png)
![4](https://user-images.githubusercontent.com/45864985/181674937-2cc4b363-4544-49f0-a9da-8d22239525cf.png)
![5](https://user-images.githubusercontent.com/45864985/181674942-c3b4cd4c-c945-45ac-8673-48e38367dc04.png)
